### ISSUE TITLE
Document extra spec enforcements on top of jsonschema

### ISSUE THREAD

**Issue by erohmensing** [airbytehq/airbyte#19725]
Currently, specs are checked against jsonschema in SAT, but jsonschema itself does not cover all of the standards that we want to enforce. We currently enforce a few things that jsonschema maintains as a `SHOULD` instead of a `MUST`. [Link to current enforcements](https://docs.airbyte.com/connector-development/connector-specification-reference#airbyte-modifications-to-jsonschema)

There may be more things we want to enforce with regards to improving the UX from the frontend side. We want to document which JSONSchema values are actually supported in the UI.

1. Both teams work together to look at current expectations of restrictions (e.g. things we think jsonschema enforces, but only suggests) and possible new ones
2. Teams collaborate on documenting these restriction in the Actor Specification docs
3. Connector Ops to write SAT tests to enforce that specs follow these enforcements

Here are some ideas of other things we may want to enforce:
* a field with `format: date` or `format: date-time` is `type: string` (and match the [expected format](https://json-schema.org/understanding-json-schema/reference/string.html#dates-and-times) via regex?)
* all items in enum lists should be of type `type` if `type` specified
* if a list has a default value, the list also needs to have a "default" option 
    * maybe that is something that the frontend should infer via there being a default value, and adding that option instead? multiple ways to do something, but looking at those rules makes us sure that we've made
* use of `date` and `datetime` according to

**Comment by octavia-squidington-iv** [comment]
cc @airbytehq/frontend

**Comment by erohmensing** [comment]
Related Issues/PRs: 
https://github.com/airbytehq/airbyte/issues/18741
https://github.com/airbytehq/airbyte/pull/19243

**Comment by evantahler** [comment]
Possibly closed by https://github.com/airbytehq/airbyte/issues/20063

**Comment by flash1293** [comment]
> a field with format: date or format: date-time is type: string (and match the [expected format](https://json-schema.org/understanding-json-schema/reference/string.html#dates-and-times) via regex?)

it will warn about the matching regex, but it doesn't check type: string , we could add that. The UI currently just ignores the format if the type is not string, so it can't break anything (worst case is that it's not respected)


> all items in enum lists should be of type type if type specified

we don't check this, however we do cast the selected value to the specified type in the UI. Could be added as well but has little potential to break anything.

> if a list has a default value, the list also needs to have a "default" option

We don't check this, that's a good call as it can cause a weird UI with a default value that has to be changed otherwise the validation fails


In summary, there are some good ideas in this issue that can be added to the existing tests!

**Comment by erohmensing** [comment]
Sounds like we could add more tests for a few things - just wondering @flash1293, are we also thinking of adding some of [these enforcements](https://github.com/airbytehq/airbyte/issues/18741) to the documentation?

@evantahler I think this is also an interesting conversation: re - spec vs metadata for running sources as containers vs via the UI. Obviously here its more about UI restrictions vs what the UI can add to the experience, but it's interesting that we have to restrict the data format a source can use in order to make it fit what the UI can handle
